pennylane/ops/identity.py

```diff
@@ -47,30 +48,8 @@ class Identity(CVObservable, Operation):
     def label(self, decimals=None, base_label=None, cache=None):
         return base_label or "I"
 
-    @staticmethod
-    def compute_eigvals():  # pylint: disable=arguments-differ
-        r"""Eigenvalues of the operator in the computational basis (static method).
-
-        If :attr:`diagonalizing_gates` are specified and implement a unitary :math:`U`,
-        the operator can be reconstructed as
-
-        .. math:: O = U \Sigma U^{\dagger},
-
-        where :math:`\Sigma` is the diagonal matrix containing the eigenvalues.
-
-        Otherwise, no particular order for the eigenvalues is guaranteed.
-
-        .. seealso:: :meth:`~.Identity.eigvals`
-
-        Returns:
-            array: eigenvalues
-
-        **Example**
-
-        >>> print(qml.Identity.compute_eigvals())
-        [ 1 1]
-        """
-        return np.array([1, 1])
+    def eigvals(self):
+        return np.ones(2 ** len(self.wires))
 
     @staticmethod
     def compute_matrix():  # pylint: disable=arguments-differ
```

Comment:
Can't remove this because then the `test_generator.py` fails.

Reply:
Why?

Reply:
We could store the number of wires as a hyperparameter and then have a call signature for `compute_matrix` that is 
```suggestion
    def compute_matrix(n_wires=1):  # pylint: disable=arguments-differ
```

Reply:
Because of this line: https://github.com/PennyLaneAI/pennylane/blob/ae8ed09562ef466aa35e372af34b104aabe2ec45/pennylane/utils.py#L104

Reply:
I like the idea that you propose, but by definition the `compute_matrix` doesn't consider wires, thus it would be a contradiction. We could just leave it as it is, meaning that `compute_matrix` returns the identity of one wire, and if the user wants the matrix of a multiple-wire identity they can use the `matrix` method.

Reply:
If we had it as an optional argument `n_wires=1`, then `i.compute_matrix()` should still work and give the one qubit case.

Reply:
Done!